moviebot/nlu/slot_annotator.py

```diff
@@ -1,286 +1,30 @@
-                        break
-            if len(params) > 0:
-                return params
-
-    def _year_annotator(self, slot, raw_utterance, utterance):
-        """
-
-        Args:
-            slot: 
-            raw_utterance: 
-            utterance: 
-
-        """
-        # fitst option is to find if any value is in the possible values
-        possible_years = [
-            int(val) for val in re.findall(r'\b\d+', raw_utterance)
-        ]
-        for year in possible_years:
-            _year = str(year)
-            if _year + 's' in raw_utterance:    # check if it's 1990s instead of
-                # 1990 or 90s
-                if len(_year) == 4:
-                    if year % 10 == 0:
-                        return [
-                            ItemConstraint(slot, Operator.BETWEEN,
-                                           f'{_year} AND {str(year + 10)}')
-                        ]
-                    else:
-                        return [ItemConstraint(slot, Operator.EQ, _year)]
-                elif len(_year) == 2:
-                    if year <= 20:
-                        _year = '20' + _year
-                        if year % 10 == 0:
-                            return [
-                                ItemConstraint(
-                                    slot, Operator.BETWEEN, f'{_year} AND'
-                                    f' {str(int(_year) + 10)}')
-                            ]
-                        else:
-                            return [ItemConstraint(slot, Operator.EQ, _year)]
-                    else:
-                        _year = '19' + _year
-                        if year % 10 == 0:
-                            return [
-                                ItemConstraint(
-                                    slot, Operator.BETWEEN, f'{_year} AND'
-                                    f' {str(int(_year) + 10)}')
-                            ]
-                        else:
-                            return [ItemConstraint(slot, Operator.EQ, _year)]
-            elif _year + 'th' in raw_utterance:    # it can be string like 19th, 20th
-                if len(_year) == 2:
-                    return [
-                        ItemConstraint(slot, Operator.BETWEEN, f'{_year}00 AND'
-                                       f' {str(year + 1)}00')
-                    ]
-            if len(_year) == 4:
-                return [ItemConstraint(slot, Operator.EQ, _year)]
-        # adding a few more elements
-        if 'new' in utterance or 'latest' in utterance:
-            return [ItemConstraint(slot, Operator.GT, '2010')]
-        if 'old' in utterance:
-            return [ItemConstraint(slot, Operator.LT, '2010')]
-
-    def find_in_raw_utterance(self, raw_utterance, gram, ngram_size):
-        """
+    @abc.abstractmethod
+    def slot_annotation(self, slot, utterance, raw_utterance):
+        """ Given utterance and a slot return a list of triplets of 
```

Comment:
Return a list of SemanticAnnotations instead.
SemanticAnnotation should be an abstract class, with ItemConstraint as a subclass of it.